hack/tools/release/notes_test.go

```diff
@@ -61,3 +61,47 @@ func Test_trimTitle(t *testing.T) {
 		})
 	}
 }
+
+func Test_trimAreaFromTitle(t *testing.T) {
+
+	tests := []struct {
+		name  string
+		title string
+		area  string
+		want  string
+	}{
+		{
+			name:  "PR title with area",
+			title: "test/e2e: improve logging for a detected rollout",
```

Comment:
this made me look at the implementation and I think it might a bit dangerous
The "search" we do is global, so if a PR title has the prefix in the middle of the string, we will "cut" from there. Instead, I think it would be better if we remove the area only when we find it at the start of the string.

example of problematic title:
`Update API:Machine with new field`
will be converted to
`Machine with new field`

Reply:
@g-gaston the area that we are removing is user friendly area https://github.com/kubernetes-sigs/cluster-api/blob/46412f0a4ea65d8f02478d2ad09ce12925485f56/hack/tools/release/notes.go#L459

Example in our case `test/e2e:` but the user friendly area is `e2e` and it makes sense to remove whole `test/e2e`

Also we are searching for area + ":" to remove , and i think it doesn't makes sense to have `area:` in between the PR title.
WDYT?

Reply:
> Also we are searching for area + ":" to remove , and i think it doesn't makes sense to have area: in between the PR title.

I don't disagree but I don't want to make that restriction for PR titles. The idea here is to cleanup PR titles for the release notes when we can, not to impose more rules over PR titles. I would prefer to restrict the cleanup to just the beginning of the title. If that means we can't automatically cleanup some titles and we do it manually, it's an ok tradeoff.

> Example in our case test/e2e: but the user friendly area is e2e and it makes sense to remove whole test/e2e:

I understand what your are saying about that PR specifically, but we can't really catch all the possible variations. What if the author starts the PR title with `Test/E2E`? We won't be able to catch that with the current logic, although it's still duplicated.

If what you want is to be able to catch composed prefixes with the area, why don't you use a regex? That way you can restrict it to start at the beginning of the title and maybe even make it case insensitive.
Something like (I haven't tested this btw, just a direction to explore)
```go
`(?mi)^\S*/e2e:\s*`
```